File: exercise/kotlin/day24/src/main/kotlin/domain/core/RegisteredRoutes.kt
package domain.core

import java.util.function.Consumer

class RegisteredRoutes {
    private var routes: Map<String, Consumer<Event>> = emptyMap()

    fun dispatch(event: Event) {
        routes[event::class.java.name]?.accept(event)
    }

    fun register(eventType: String, apply: Consumer<Event>) {
        routes = routes + (eventType to apply)
    }
}

File: exercise/python/day24/domain/core/registered_routes.py
from typing import Dict, Callable

from domain.core.event import Event


class RegisteredRoutes:
    def __init__(self):
        self.routes: Dict[str, Callable[[Event], None]] = {}

    def dispatch(self, event: Event):
        if event.__class__.__name__ in self.routes:
            self.routes[event.__class__.__name__](event)

    def register(self, event_type: str, apply: Callable[[Event], None]):
        self.routes[event_type] = apply
